File: backend/DAL/Entities/DangerMessage.cs
namespace DAL.Entities;

public class DangerMessage : Entity {
  public string Message { get; set; }

  public Guid UserId { get; set; }

  public Guid DangerId { get; set; }

  public Guid? ReferencedMessageId { get; set; }

  // Foreign Keys
  public User User { get; set; }
  public Danger Danger { get; set; }
  public DangerMessage? ReferencedMessage { get; set; }

  // Reverse Navigation
  public ICollection<DangerMessage> Answers { get; set; }
}

File: frontend/src/services/entities/DangerMessage.ts
import { Danger } from "./Danger";
import { User } from "./User";


export interface DangerMessage {
    id?: string;
    createdAt?: Date;
    updatedAt?: Date | undefined;
    message?: string | undefined;
    userId?: string;
    dangerId?: string;
    referencedMessageId?: string | undefined;
    user?: User;
    danger?: Danger;
    referencedMessage?: DangerMessage;
    answers?: DangerMessage[] | undefined;
}
